src/osd/OSD.cc

```diff
@@ -5846,10 +5846,20 @@ void OSD::handle_scrub(MOSDScrub *m)
       PG *pg = p->second;
       pg->lock();
       if (pg->is_primary()) {
+        if (m->stop_scrub && !pg->scrubber.must_scrub) {
+	  derr << __func__ << *pg << " stop_scrub should stop manual scrub" << dendl;
+          pg->unlock();
+          continue;
+        }
 	pg->unreg_next_scrub();
```

Comment:
I wouldn't do unreg_next_scrub()/reg_next_scrub() in the stop case in both places here.  It is confusing because it is a no-op with must_scrub set.